modules/oadp-configuring-aws-bsl.adoc

```diff
@@ -0,0 +1,70 @@
+<6> The name of the secret containing the credentials.
+<7> The AWS region where the bucket is located. Optional if s3ForcePathStyle is false.
+<8> A boolean flag to decide whether to use path-style addressing instead of virtual hosted bucket addressing. Set to "true" if using a storage service such as MinIO or NooBaa. This is an optional field. The default value is `false`.
+<9> You can specify the {aws-short} S3 URL here for explicitness. This field is primarily for storage services such as MinIO or NooBaa. This is an optional field.
+<10> This field is primarily used for storage services such as MinIO or NooBaa. This is an optional field.
+<11> The name of the server-side encryption algorithm to use for uploading objects, for example, `AES256`. This is an optional field.
+<12> Specify an {aws-short} KMS key ID. You can format, as shown in the example, as an alias, such as `alias/<KMS-key-alias-name>`, or the full `ARN` to enable encryption of the backups stored in S3. Note that, `kmsKeyId` cannot be used in with `customerKeyEncryptionFile`. This is an optional field.
+<13> Specify the file that has the `SSE-C` customer key to enable customer key encryption of the backups stored in S3. The file must contain a 32-byte string. The `customerKeyEncryptionFile` points to a mounted secret within the `velero` container. Add the following key-value pair to the `velero` `cloud-credentials` secret: `customer-key: <your_b64_encoded_32byte_string>`. Note that the `customerKeyEncryptionFile` field cannot be used with the `kmsKeyId`. The default value is an empty string (""), which means, `SSE-C` is disabled. This is an optional field.
+<14> The version of the signature algorithm used to create signed URLs. You use signed URLs to download the backups, or fetch the logs. Possible versions are `1` and `4`. The default version is `4`. This is an optional field.
+<15> The name of the {aws-short} profile in the credentials file. Default value is `default`. This is an optional field.
+<16> Set the `insecureSkipTLSVerify` field to `true` if you do not want to verify the TLS certificate when connecting to the object store, for example, for self-signed certificates with MinIO. Setting to `true` is susceptible to man-in-the-middle attacks and is not recommended for production workloads. Default value is `false`. This is an optional field.
+<17> Set the `enableSharedConfig` field to `true` if you want to load the credentials file as a shared config file. Default value is `false`. This is an optional field.
+<18> Specify the tags to annotate the {aws-short} S3 objects. Specify the tags in key-value pairs. Default value is an empty string (""). This is an optional field.
+<19> Specify the checksum algorithm to use for uploading objects to S3. The supported values are: `CRC32`, `CRC32C`, `SHA1`, and `SHA256`. If you set the field as an empty string (""), the checksum check will be skipped. Default value is `CRC32`. This is an optional field.
```

Comment:
Just a few things that I've mentioned earlier in this review:
```suggestion
<17> Set the `enableSharedConfig` field to `true` if you want to load the credentials file as a shared config file. The default value is `false`. This is an optional field.
<18> Specify the tags to annotate the {aws-short} S3 objects. Specify the tags in key-value pairs. The default value is an empty string (`""`). This is an optional field.
<19> Specify the checksum algorithm to use for uploading objects to S3. The supported values are: `CRC32`, `CRC32C`, `SHA1`, and `SHA256`. If you set the field as an empty string (`""`), the checksum check will be skipped. The default value is `CRC32`. This is an optional field.
```

Reply:
Done.